test/mongoose_cowboy_metrics_SUITE.erl

```diff
@@ -0,0 +1,201 @@
+    [ensure_metric_value(response_latency_metric(?DEFAULT_PREFIX, M, C), 0) || M <- ?METHODS, C <- ?CLASSES -- [<<"2XX">>]],
+
+    stop_listener(),
+    destroy_metrics().
+
+%%-------------------------------------------------------------------
+%% cowboy handler callbacks
+%%-------------------------------------------------------------------
+
+init(_Type, Req, Opts) ->
+    {ok, Req, proplists:get_value(action, Opts, echo_status)}.
+
+handle(Req, echo_status = State) ->
+    {BinStatus, Req1} = cowboy_req:qs_val(<<"status">>, Req),
+    Status = binary_to_integer(BinStatus),
+    {ok, RespReq} = cowboy_req:reply(Status, [], <<>>, Req1),
+    {ok, RespReq, State};
+handle(Req, none = State) ->
+    {ok, Req, State}.
+
+terminate(_Reason, _Req, _State) ->
+    ok.
+
+%%-------------------------------------------------------------------
+%% Helpers
+%%-------------------------------------------------------------------
+
+start_listener(Paths, ExtraEnv) ->
+    Dispatch = cowboy_router:compile([{'_', Paths}]),
+    Middleware = [mongoose_cowboy_metrics_mw_before,
+                  cowboy_router,
+                  mongoose_cowboy_metrics_mw_after,
+                  cowboy_handler],
+    {ok, _} = cowboy:start_http(?LISTENER, 1, [{port, 0}],
+                                [{env, [{dispatch, Dispatch} | ExtraEnv]},
+                                 {middlewares, Middleware}]).
+
+stop_listener() ->
+    ok = cowboy:stop_listener(?LISTENER).
+
+create_metrics() ->
+    create_metrics(?DEFAULT_PREFIX).
+
+create_metrics(Prefix) ->
+    %% in case of request count we're interested in the exact values - let's use counter
+    [ok = mongoose_metrics:ensure_metric(global, M, counter) || M <- count_metrics(Prefix)],
+    %% in case of latencies it doesn't make sense to check if values are correct - we
+    %% only need to know if the value has been reported, so gauge is enough
+    [ok = mongoose_metrics:ensure_metric(global, M, gauge) || M <- latency_metrics(Prefix)].
+
+destroy_metrics() ->
+    destroy_metrics(?DEFAULT_PREFIX).
+
+destroy_metrics(Prefix) ->
+    [exometer:delete([global | M]) || M <- count_metrics(Prefix) ++ latency_metrics(Prefix)].
+
+count_metrics(Prefix) ->
+    [request_count_metric(Prefix, M) || M <- ?METHODS] ++
+    [response_count_metric(Prefix, M, C) || M <- ?METHODS, C <- ?CLASSES].
+
+latency_metrics(Prefix) ->
+    [response_latency_metric(Prefix, M, C) || M <- ?METHODS, C <- ?CLASSES].
+
+ensure_metric_value(Metric, Value) ->
+    {ok, DataPoints} = mongoose_metrics:get_metric_value([global | Metric]),
+    ?assertEqual(Value, proplists:get_value(value, DataPoints)).
+
+ensure_metric_bumped(Metric) ->
+    {ok, DataPoints} = mongoose_metrics:get_metric_value([global | Metric]),
+    ?assertNotEqual(0, proplists:get_value(value, DataPoints)).
+
+run_requests(Methods, Statuses) ->
+    {ok, Conn} = gun:open("localhost", ranch:get_port(?LISTENER)),
```

Comment:
In small tests we already use `fusco` when there is a need to send an HTTP request. It'd better to re-use what is already used, in my opinion. I do agree that there are better libs than `fusco`. At some point it will be replaced by sth else (`gun` or sth else).

Reply:
😢